oneview_redfish_toolkit/blueprints/resource_block.py

```diff
@@ -199,25 +199,35 @@ def _get_oneview_resource(uuid):
 
         resource = getattr(g.oneview_client, cached_categ.resource)
         function = getattr(resource, cached_categ.function)
+        result = function(resource_uuid)
+        if isinstance(result, dict) :
+            return result
+        else:
+            return result.data
+
 
-        return function(resource_uuid)
 
     categories = [
-        {"func": g.oneview_client.server_hardware.get, "param": uuid},
-        {"func": g.oneview_client.server_profile_templates.get, "param": uuid},
+        {"func": g.oneview_client.server_hardware.get_by_id, "param": uuid},
+        {"func": g.oneview_client.server_profile_templates.get_by_id, "param": uuid},
         {"func": g.oneview_client.index_resources.get, "param": drives_param},
-        {"func": g.oneview_client.volumes.get, "param": uuid}
+        {"func": g.oneview_client.volumes.get_by_id, "param": uuid}
```

Comment:
For index_resources get call available?

Reply:
Index call is not yet upgraded and hence its does not have get_by_id or by URI